docs/guides/interfaces.rst

```diff
@@ -1,10 +1,89 @@
-.. _interfaces_guide:
+.. _interfaces:
+========================
+Interfaces
+========================
 
-===================
-Interfaces guide
-===================
 
-This guide is currently a work in progress and only contains bullet points for now.
+One of the biggest strengths of Orquestra is its modularity. Integrating new backends, optimizers, ansatzes, etc. does not require changing the core code - it requires only creating a new module that conforms to existing interfaces and therefore can be used across the whole platform. Using our interfaces help you add whatever building blocks you need for your project and ensure that they will work well with other existing elements.
 
-Cost Functions in VQAs
-======================
\ No newline at end of file
+What is an interface?
+=====================
+
+Before we get into the details, let's explain what we mean by "interfaces" in this context. We use this word as a shorthand for "a way to implement certain concepts in such a way, that as long as two implementations conform to a particular interface, they can be swapped and the program should still work correctly". Obviously, various implementations might work differently - e.g.: one simulator might be faster than the other - but they should be undistinguishable when it comes to inputs, outputs and how they are being used.
```

Comment:
I think that the correct word is `indistinguishable`, see e.g. discussion [here](https://english.stackexchange.com/questions/3544/undistinguishable-vs-indistinguishable): 

```suggestion
Before we get into the details, let's explain what we mean by "interfaces" in this context. We use this word as a shorthand for "a way to implement certain concepts in such a way, that as long as two implementations conform to a particular interface, they can be swapped and the program should still work correctly". Obviously, various implementations might work differently - e.g.: one simulator might be faster than the other - but they should be indistinguishable when it comes to inputs, outputs and how they are being used.
```